docs/Configuring-Orion/TLS.md

```diff
@@ -3,15 +3,162 @@ description: TLS
+
+1. Generate a key file called orion_cer.key by running the following command:
+
+    `openssl genrsa -out orion_cer.key 2048`
+  
+1. Generate a certificate signing request (CSR) called orion_cer.csr by running the following command:
+
+    `openssl req -new -key orion_cer.key -out orion_cer.csr`
+  
+1. Answer each prompt for information that will be added to the certificate request. Ensure that the
+value you specify for Common Name (CN) matches the host public DNS so that the requests from the server are accepted. The name 
+is used in the configuration file for the `nodeurl` and `clienturl` options too.
+
+1. Generate a certificate called orion_cer.pem signed by the CA certificate by runnig the following command:
+
+    `openssl x509 -req -in orion_cer.csr -CA orion_ca.pem -CAkey orion_ca.key -CAcreateserial -out orion_cer.pem -days 500 -sha256`
+
+#### Configuration file example that uses TLS and the created certificates
+
+```
+nodeurl = "<YOUR-COMMON-NAME>:8080/"
+nodeport = 8080
+
+clienturl = "<YOUR-COMMON-NAME>:8888/"
+clientport = 8888
+
+othernodes = ["<YOUR-COMMON-NAME>:8080"]
+
+workdir = "<PATH-TO>/orion/bin"
+
+publickeys = ["orion.pub"]
+privatekeys = ["orion.key"]
+
+tls = "strict"
+
+tlsservercert = "<PATH-TO>/orion/bin/orion_cer.pem"
+tlsserverchain = ["<PATH-TO>/orion/bin/orion_ca.pem"]
+tlsserverkey = "<PATH-TO>/orion/bin/orion_cer.key"
+
+tlsclientcert = "<PATH-TO>/orion/bin/orion_cer.pem"
+tlsclientchain = ["<PATH-TO>/orion/bin/orion_ca.pem"]
+tlsclientkey = "<PATH-TO>/orion/bin/orion_cer.key"
+
+tlsclienttrust = "ca"
+tlsservertrust = "ca"
+
+nodenetworkinterface = "0.0.0.0"
+clientnetworkinterface = "0.0.0.0"
+```
+
+### Generating certificates when the public IP address is the Common Name (CN) 
+
+If you want to use a public IP as the CN, complete the following steps:
+
+#### Updating the `openssl.cfn` file
+
+1. Find the `openssl.cfn` file, and create a copy of it.
+1. In your copy of the `openssl.cfn` file, find the `[req]` section, and add the following lines:
+
+  ```
+  req_extensions = v3_req
+  
+  [ v3_req ]
+  basicConstraints = CA:FALSE
+  keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+  subjectAltName = @alt_names
+
+  [alt_names]
+  DNS.1 = <DNS-PUBLIC-RECORD>
+  DNS.2 = <DNS-PRIVATE-RECORD>
+  IP.1 = <PUBLIC-IP-ADDRESS>
+  IP.2 = <PRIVATE-IP-ADDRESS>
+  ```
+
+1. For each DNS that you want to use as an alternate name, specify a DNS._n_ entry.
+1. For each IP address that you want as an alternate IP address, specify an IP._n_ entry.
+
+#### Generating a new CSR for a node
+
+1. Run the following command. Ensure you substitute your values for all variables.
+
+  `openssl req -new -key orion_cer.key -out orion_cer.csr -config <PATH-TO>/openssl.cnf`
+
+1. Test whether the certificate was generated with the expected subject alternative names by running the following command:
+
+  `openssl req -text -noout -in orion_cer.csr`
```

Comment:
Same issue as the code block above -> this isn't indented in the published output and the numbering restarts at 1 underneath this